library/proc_macro/src/lib.rs

```diff
@@ -325,45 +330,22 @@ impl ConcatStreamsHelper {
     }
 }
 
-/// Collects a number of token trees into a single stream.
-#[stable(feature = "proc_macro_lib2", since = "1.29.0")]
-impl FromIterator<TokenTree> for TokenStream {
-    fn from_iter<I: IntoIterator<Item = TokenTree>>(trees: I) -> Self {
-        let iter = trees.into_iter();
-        let mut builder = ConcatTreesHelper::new(iter.size_hint().0);
-        iter.for_each(|tree| builder.push(tree));
-        builder.build()
-    }
-}
-
-/// A "flattening" operation on token streams, collects token trees
-/// from multiple token streams into a single stream.
-#[stable(feature = "proc_macro_lib", since = "1.15.0")]
-impl FromIterator<TokenStream> for TokenStream {
-    fn from_iter<I: IntoIterator<Item = TokenStream>>(streams: I) -> Self {
-        let iter = streams.into_iter();
+#[stable(feature = "proc_macro_totokens_migration", since = "CURRENT_RUSTC_VERSION")]
```

Comment:
Since trait impl is insta-stable, is it the proper thing to directly mark the migration as `#[stable]` here (with another feature gate name) and remove the old trait impl?

Reply:
I do not think we would want to add this impl until after stabilization of the trait. The process of stabilizing the trait involves agreeing on what types should implement it. Whereas this impl immediately freezes the current tentative set of types as a permanent commitment.

Reply:
I see, I have reverted the changes about the impl migration.